src/crimson/os/alienstore/alien_store.cc

```diff
@@ -363,17 +363,17 @@ seastar::future<> AlienStore::do_transaction(CollectionRef ch,
     std::move(done),
     [this, ch, id] (auto &txn, auto &done) {
       return seastar::with_gate(transaction_gate, [this, ch, id, &txn, &done] {
-	return tp_mutex.lock().then ([this, ch, id, &txn, &done] {
+	AlienCollection* alien_coll = static_cast<AlienCollection*>(ch.get());
+	return alien_coll->with_lock([this, ch, id, &txn, &done] {
 	  Context *crimson_wrapper =
 	    ceph::os::Transaction::collect_all_contexts(txn);
 	  return tp->submit([this, ch, id, crimson_wrapper, &txn, &done] {
 	    txn.register_on_commit(new OnCommit(id, done, crimson_wrapper, txn));
 	    auto c = static_cast<AlienCollection*>(ch.get());
 	    return store->queue_transaction(c->collection, std::move(txn));
 	  });
-	}).then([this, &done] (int r) {
+	}).then([this, alien_coll, &done] (int r) {
```

Comment:
could drop `alien_coll`, which is not used anymore in the latest revision.

Reply:
@tchaikov Sorry for the inconvenience, just corrected it:-)

Reply:
no worries. thank you!